### PR TITLE
optimize: softlink the plugin to $GOPATH/bin

### PR DESCRIPTION
**Pull request by FGYFFFF**
#### What type of PR is this?

optimize
#### Check the PR title.

- [x] This PR title match the format: \<type\>(optional scope): \<description\>
- [x] The description of this PR title is user-oriented and clear enough for others to understand.

#### (Optional) Translate the PR title into Chinese.
将插件软链接到 $GOPATH/bin 下

#### (Optional) More detail description for this PR(en: English/zh: Chinese).

en: softlink the plugin to $GOPATH/bin
zh(optional): 将插件软链接到 $GOPATH/bin 下

### CODE REVIEW

**Comment by FGYFFFF** [comment]
Previously, it would softlink the protoc plugins to different locations depending on the installation location of protoc, resulting in a possible insufficient permissions problem

File: cmd/hz/internal/config/cmd.go [hunk 1]
```diff
@@ -58,7 +58,12 @@ func lookupTool(idlType string) (string, error) {
 	if err != nil {
 		return "", fmt.Errorf("failed to get executable path: %s", err)
 	}
-	dir := filepath.Dir(path)
+	gopath, err := util.GetGOPATH()
```
Comment (welkeyever) [inline cmd/hz/internal/config/cmd.go:61]: 上面42行那里也在获取GOPATH？这个行为可以统一一下？

File: cmd/hz/internal/config/cmd.go [hunk 2]
```diff
@@ -58,7 +58,12 @@ func lookupTool(idlType string) (string, error) {
 	if err != nil {
 		return "", fmt.Errorf("failed to get executable path: %s", err)
 	}
-	dir := filepath.Dir(path)
```
Comment (welkeyever) [inline cmd/hz/internal/config/cmd.go:61]: 之前的这个`path`里是啥

File: cmd/hz/internal/config/cmd.go (hunk 1 above)
Comment (FGYFFFF) [reply cmd/hz/internal/config/cmd.go:61]: 上面那段逻辑有点问题，[179](https://github.com/cloudwego/hertz/pull/179) 已经给修了，确实可以复用下

File: cmd/hz/internal/config/cmd.go (hunk 2 above)
Comment (FGYFFFF) [reply cmd/hz/internal/config/cmd.go:61]: 之前这里的path根据"protoc"/"thriftgo"的位置而定，他们在哪插件就放哪里，可能有权限问题

File: cmd/hz/internal/config/cmd.go [hunk 3]
```diff
@@ -58,7 +58,12 @@ func lookupTool(idlType string) (string, error) {
...
+	gopath, err := util.GetGOPATH()
+	if err != nil {
+		return "", err
+	}
+	// softlink the plugin to "$GOPATH/bin"
+	dir := gopath + "/bin"
```
Comment (welkeyever) [inline cmd/hz/internal/config/cmd.go:66]: 这个跨平台（类似windows）上有问题吗

Comment (FGYFFFF) [reply cmd/hz/internal/config/cmd.go:66]: 没问题，这块正确逻辑就是要放到"$GOPATH/bin"下面的

Comment (welkeyever) [reply cmd/hz/internal/config/cmd.go:66]: windows下的文件分隔符不是 \ 吗

Comment (FGYFFFF) [reply cmd/hz/internal/config/cmd.go]: done

**Review by welkeyever — APPROVED** [review]

**Review by Duslia — APPROVED** [review]

**Pull request merged**